esx_service/utils/vmdk_utils.py

```diff
@@ -305,6 +305,16 @@ def log_volume_lsof(vol_name):
                 cartel, name, ftype, fd, desc)
             logging.info("Volume open descriptor: %s", msg)
 
+def get_datastore_url(datastore):
+    si = vmdk_ops.get_si()
+    res = [d.info.url for d in si.content.rootFolder.childEntity[0].datastore if d.info.name == datastore]
```

Comment:
pls wrap `si.content.rootFolder.childEntity[0].datastore` in some named function and reuse in line 315 as well